// @progress/kendo-angular-grid/dist/es2015/filtering/cell/filter-cell-operators.component.js
import * as tslib_1 from "tslib";
import { Component, Input, Output, EventEmitter, HostBinding, ViewChild } from '@angular/core';
import { DropDownListComponent } from '@progress/kendo-angular-dropdowns';
/**
 * Represents a component which accommodates the filter operators.
 */
let FilterCellOperatorsComponent = class FilterCellOperatorsComponent {
    /**
     * Represents a component which accommodates the filter operators.
     */
    constructor() {
        /**
         * The filter operators that will be shown.
         */
        this.operators = [];
        /**
         * Determines if the list of operators will be shown.
         * @type {boolean}
         */
        this.showOperators = true;
        /**
         * Fires when the operator is selected.
         * @type {EventEmitter<string>}
         */
        this.valueChange = new EventEmitter();
        /**
         * Fires when the **Clear** button is clicked.
         * @type {EventEmitter<{}>}
         */
        this.clear = new EventEmitter();
    }
    /**
     * @hidden
     */
    get hostClasses() {
        return true;
    }
    /**
     * @hidden
     */
    onChange(dataItem) {
        this.valueChange.emit(dataItem);
    }
    /**
     * @hidden
     */
    clearClick() {
        this.clear.emit();
    }
    /**
     * @hidden
     */
    clearKeydown(args) {
        if (args.keyCode === 13 /* enter */ || args.keyCode === 32 /* space */) {
            this.clear.emit();
        }
    }
    /**
     * @hidden
     */
    dropdownKeydown(args) {
        if (args.defaultPrevented) {
            return;
        }
        if (args.keyCode === 13 /* enter */ && !this.dropdown.isOpen) {
            this.dropdown.toggle(true);
            args.preventDefault();
        }
    }
};
tslib_1.__decorate([
    HostBinding('class.k-filtercell-operator'),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [])
], FilterCellOperatorsComponent.prototype, "hostClasses", null);
tslib_1.__decorate([
    ViewChild('dropdown'),
    tslib_1.__metadata("design:type", DropDownListComponent)
], FilterCellOperatorsComponent.prototype, "dropdown", void 0);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Array)
], FilterCellOperatorsComponent.prototype, "operators", void 0);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Boolean)
], FilterCellOperatorsComponent.prototype, "showButton", void 0);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Boolean)
], FilterCellOperatorsComponent.prototype, "showOperators", void 0);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String)
], FilterCellOperatorsComponent.prototype, "value", void 0);
tslib_1.__decorate([
    Output(),
    tslib_1.__metadata("design:type", EventEmitter)
], FilterCellOperatorsComponent.prototype, "valueChange", void 0);
tslib_1.__decorate([
    Output(),
    tslib_1.__metadata("design:type", EventEmitter)
], FilterCellOperatorsComponent.prototype, "clear", void 0);
FilterCellOperatorsComponent = tslib_1.__decorate([
    Component({
        selector: 'kendo-grid-filter-cell-operators',
        template: `
        <kendo-dropdownlist
            #dropdown
            *ngIf="showOperators"
            kendoGridFocusable
            [data]="operators"
            class="k-dropdown-operator"
            (valueChange)="onChange($event)"
            [value]="value"
            [iconClass]="'k-i-filter'"
            [valuePrimitive]="true"
            [textField]="'text'"
            [popupSettings]="{ width: 'auto' }"
            [valueField]="'value'"
            (keydown)="dropdownKeydown($event)">
        </kendo-dropdownlist>
        <button
            kendoGridFocusable
            [ngClass]="{'k-clear-button-visible': showButton}"
            class="k-button k-button-icon"
            title="Clear"
            (click)="clearClick()"
            (keydown)="clearKeydown($event)">
                <span class="k-icon k-i-filter-clear"></span>
        </button>
    `
    })
], FilterCellOperatorsComponent);
export { FilterCellOperatorsComponent };
