// @progress/kendo-angular-grid/dist/es2015/filtering/cell/filter-cell.component.js
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
import { ColumnComponent } from '../../columns/column.component';
import { isPresent, isNullOrEmptyString } from '../../utils';
/**
 * @hidden
 */
let FilterCellComponent = class FilterCellComponent {
    /**
     * @hidden
     */
    constructor() {
        this._templateContext = {};
    }
    get templateContext() {
        this._templateContext.column = this.column;
        this._templateContext.filter = this.filter;
        // tslint:disable-next-line:no-string-literal
        this._templateContext["$implicit"] = this.filter;
        return this._templateContext;
    }
    get hasTemplate() {
        return isPresent(this.column.filterCellTemplateRef);
    }
    get isFilterable() {
        return isPresent(this.column) && !isNullOrEmptyString(this.column.field) && this.column.filterable;
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", ColumnComponent)
], FilterCellComponent.prototype, "column", void 0);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Object)
], FilterCellComponent.prototype, "filter", void 0);
FilterCellComponent = tslib_1.__decorate([
    Component({
        selector: '[kendoGridFilterCell]',
        template: `
        <ng-template [ngIf]="isFilterable">
            <ng-container [ngSwitch]="hasTemplate">
                <ng-container *ngSwitchCase="false">
                    <ng-container kendoFilterCellHost [column]="column" [filter]="filter"></ng-container>
                </ng-container>
                <ng-container *ngSwitchCase="true">
                    <ng-template
                        *ngIf="column.filterCellTemplateRef"
                        [ngTemplateOutlet]="column.filterCellTemplateRef"
                        [ngTemplateOutletContext]="templateContext">
                    </ng-template>
                </ng-container>
            </ng-container>
        </ng-template>
    `
    })
], FilterCellComponent);
export { FilterCellComponent };
