// @progress/kendo-angular-grid/dist/es2015/filtering/boolean-filter.component.js
// tslint:disable:no-access-missing-member
import * as tslib_1 from "tslib";
import { Input, HostBinding } from '@angular/core';
import { ColumnComponent } from '../columns/column.component';
import { BaseFilterCellComponent } from './base-filter-cell.component';
/**
 * @hidden
 */
export class BooleanFilterComponent extends BaseFilterCellComponent {
    constructor(filterService, localization) {
        super(filterService);
        this.localization = localization;
        /**
         * @hidden
         */
        this.operator = "eq";
        /**
         * @hidden
         */
        this.items = [
            { text: this.localization.get("filterIsTrue"), value: true },
            { text: this.localization.get("filterIsFalse"), value: false }
        ];
        /**
         * @hidden
         */
        this.defaultItem = { text: this.localization.get("filterBooleanAll"), value: null };
    }
    /**
     * @hidden
     */
    get hostClasses() {
        return true;
    }
    /**
     * The current filter for the associated column field.
     * @readonly
     * @type {FilterDescriptor}
     */
    get currentFilter() {
        return this.filterByField(this.column.field);
    }
    /**
     * The current filter operator for the associated column field.
     * @readonly
     * @type {string}
     */
    get currentOperator() {
        return this.currentFilter ? this.currentFilter.operator : this.operator;
    }
}
tslib_1.__decorate([
    HostBinding('class.k-filtercell-boolean'),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [])
], BooleanFilterComponent.prototype, "hostClasses", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", ColumnComponent)
], BooleanFilterComponent.prototype, "column", void 0);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Object)
], BooleanFilterComponent.prototype, "filter", void 0);
