// @progress/kendo-angular-inputs/dist/es/slider/slider-ticks.component.js
import * as tslib_1 from "tslib";
import { Component, Inject, Input, Optional, QueryList, ViewChildren, HostBinding } from '@angular/core';
import { RTL } from '@progress/kendo-angular-l10n';
import { SliderUtil as Util } from '@telerik/kendo-inputs-common';
/**
 * @hidden
 */
var SliderTick = /** @class */ (function () {
    function SliderTick(value) {
        this.value = value;
        this.classes = {
            'k-tick': true
        };
    }
    return SliderTick;
}());
/* tslint:disable:component-selector */
/**
 * @hidden
 */
var SliderTicksComponent = /** @class */ (function () {
    function SliderTicksComponent(rtl) {
        this.rtl = rtl;
        this.wrapperClasses = 'k-reset k-slider-items';
        this.ticks = [];
    }
    SliderTicksComponent.prototype.ngOnChanges = function (_) {
        this.createTicks();
    };
    SliderTicksComponent.prototype.createTicks = function () {
        var count = Util.calculateTicksCount(this.max, this.min, this.step);
        var tickValueProps = {
            min: this.min,
            max: this.max,
            smallStep: this.step
        };
        var result = [];
        for (var i = 0; i < count; i++) {
            result.push(new SliderTick(Util.calculateValueFromTick(i, tickValueProps)));
        }
        if (this.rtl || this.vertical) {
            result = result.reverse();
        }
        if (result.length > 0) {
            Object.assign(result[0].classes, this.endTickClasses(true));
            Object.assign(result[result.length - 1].classes, this.endTickClasses(false));
        }
        this.ticks = result;
    };
    SliderTicksComponent.prototype.endTickClasses = function (first) {
        return {
            'k-first': (first && !this.vertical) || (!first && this.vertical),
            'k-last': (!first && !this.vertical) || (first && this.vertical)
        };
    };
    tslib_1.__decorate([
        HostBinding('class'),
        tslib_1.__metadata("design:type", String)
    ], SliderTicksComponent.prototype, "wrapperClasses", void 0);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Function)
    ], SliderTicksComponent.prototype, "tickTitle", void 0);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean)
    ], SliderTicksComponent.prototype, "vertical", void 0);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number)
    ], SliderTicksComponent.prototype, "step", void 0);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number)
    ], SliderTicksComponent.prototype, "min", void 0);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number)
    ], SliderTicksComponent.prototype, "max", void 0);
    tslib_1.__decorate([
        ViewChildren('tickElement'),
        tslib_1.__metadata("design:type", QueryList)
    ], SliderTicksComponent.prototype, "tickElements", void 0);
    SliderTicksComponent = tslib_1.__decorate([
        Component({
            selector: '[kendoSliderTicks]',
            template: "\n    <li #tickElement *ngFor=\"let tick of ticks;\"\n        [ngClass]=\"tick.classes\"\n        title=\"{{ tickTitle(tick.value) }}\"\n        role=\"presentation\"\n     >&nbsp;</li>\n  "
        }),
        tslib_1.__param(0, Optional()), tslib_1.__param(0, Inject(RTL)),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], SliderTicksComponent);
    return SliderTicksComponent;
}());
export { SliderTicksComponent };
