// @progress/kendo-angular-grid/dist/es/filtering/operators/after-eq-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { FilterOperatorBase } from './filter-operator.base';
import { LocalizationService } from '@progress/kendo-angular-l10n';
/*
 * Represents the `GreaterOrEqualTo` (**Is after or equal to**) date filter operator.
 */
var AfterEqFilterOperatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(AfterEqFilterOperatorComponent, _super);
    function AfterEqFilterOperatorComponent(localization) {
        return _super.call(this, "after-eq", localization) || this;
    }
    AfterEqFilterOperatorComponent_1 = AfterEqFilterOperatorComponent;
    /**
     * @hidden
     */
    AfterEqFilterOperatorComponent.prototype.toJSON = function () {
        return {
            text: this.text,
            value: "gte"
        };
    };
    AfterEqFilterOperatorComponent = AfterEqFilterOperatorComponent_1 = tslib_1.__decorate([
        Component({
            providers: [
                {
                    provide: FilterOperatorBase,
                    useExisting: forwardRef(function () { return AfterEqFilterOperatorComponent_1; }) // tslint:disable-line:no-forward-ref
                }
            ],
            selector: 'kendo-filter-after-eq-operator',
            template: ""
        }),
        tslib_1.__metadata("design:paramtypes", [LocalizationService])
    ], AfterEqFilterOperatorComponent);
    return AfterEqFilterOperatorComponent;
    var AfterEqFilterOperatorComponent_1;
}(FilterOperatorBase));
export { AfterEqFilterOperatorComponent };
