// @progress/kendo-angular-grid/dist/es/filtering/operators/after-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { FilterOperatorBase } from './filter-operator.base';
import { LocalizationService } from '@progress/kendo-angular-l10n';
/*
 * Represents the `Greater` (**Is after**) date filter operator.
 */
var AfterFilterOperatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(AfterFilterOperatorComponent, _super);
    function AfterFilterOperatorComponent(localization) {
        return _super.call(this, "after", localization) || this;
    }
    AfterFilterOperatorComponent_1 = AfterFilterOperatorComponent;
    /**
     * @hidden
     */
    AfterFilterOperatorComponent.prototype.toJSON = function () {
        return {
            text: this.text,
            value: "gt"
        };
    };
    AfterFilterOperatorComponent = AfterFilterOperatorComponent_1 = tslib_1.__decorate([
        Component({
            providers: [
                {
                    provide: FilterOperatorBase,
                    useExisting: forwardRef(function () { return AfterFilterOperatorComponent_1; }) // tslint:disable-line:no-forward-ref
                }
            ],
            selector: 'kendo-filter-after-operator',
            template: ""
        }),
        tslib_1.__metadata("design:paramtypes", [LocalizationService])
    ], AfterFilterOperatorComponent);
    return AfterFilterOperatorComponent;
    var AfterFilterOperatorComponent_1;
}(FilterOperatorBase));
export { AfterFilterOperatorComponent };
