// @progress/kendo-angular-grid/dist/es/filtering/operators/contains-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { LocalizationService } from '@progress/kendo-angular-l10n';
import { FilterOperatorBase } from './filter-operator.base';
/**
 * Represents the `Contains` (**Contains**) filter operator.
 */
var ContainsFilterOperatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(ContainsFilterOperatorComponent, _super);
    function ContainsFilterOperatorComponent(localization) {
        return _super.call(this, "contains", localization) || this;
    }
    ContainsFilterOperatorComponent_1 = ContainsFilterOperatorComponent;
    ContainsFilterOperatorComponent = ContainsFilterOperatorComponent_1 = tslib_1.__decorate([
        Component({
            providers: [
                {
                    provide: FilterOperatorBase,
                    useExisting: forwardRef(function () { return ContainsFilterOperatorComponent_1; }) // tslint:disable-line:no-forward-ref
                }
            ],
            selector: 'kendo-filter-contains-operator',
            template: ""
        }),
        tslib_1.__metadata("design:paramtypes", [LocalizationService])
    ], ContainsFilterOperatorComponent);
    return ContainsFilterOperatorComponent;
    var ContainsFilterOperatorComponent_1;
}(FilterOperatorBase));
export { ContainsFilterOperatorComponent };
