// @progress/kendo-angular-grid/dist/es/filtering/operators/ends-with-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { LocalizationService } from '@progress/kendo-angular-l10n';
import { FilterOperatorBase } from './filter-operator.base';
/**
 * Represents the `EndsWith` (**Ends with**) string filter operator.
 */
var EndsWithFilterOperatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(EndsWithFilterOperatorComponent, _super);
    function EndsWithFilterOperatorComponent(localization) {
        return _super.call(this, "endswith", localization) || this;
    }
    EndsWithFilterOperatorComponent_1 = EndsWithFilterOperatorComponent;
    EndsWithFilterOperatorComponent = EndsWithFilterOperatorComponent_1 = tslib_1.__decorate([
        Component({
            providers: [
                {
                    provide: FilterOperatorBase,
                    useExisting: forwardRef(function () { return EndsWithFilterOperatorComponent_1; }) // tslint:disable-line:no-forward-ref
                }
            ],
            selector: 'kendo-filter-endswith-operator',
            template: ""
        }),
        tslib_1.__metadata("design:paramtypes", [LocalizationService])
    ], EndsWithFilterOperatorComponent);
    return EndsWithFilterOperatorComponent;
    var EndsWithFilterOperatorComponent_1;
}(FilterOperatorBase));
export { EndsWithFilterOperatorComponent };
