// @progress/kendo-angular-grid/dist/es/filtering/operators/gte-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { FilterOperatorBase } from './filter-operator.base';
import { LocalizationService } from '@progress/kendo-angular-l10n';
/**
 * Represents the `GreaterOrEqualTo` (**Is greater than or equal to**) numeric filter operator.
 */
var GreaterOrEqualToFilterOperatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(GreaterOrEqualToFilterOperatorComponent, _super);
    function GreaterOrEqualToFilterOperatorComponent(localization) {
        return _super.call(this, "gte", localization) || this;
    }
    GreaterOrEqualToFilterOperatorComponent_1 = GreaterOrEqualToFilterOperatorComponent;
    GreaterOrEqualToFilterOperatorComponent = GreaterOrEqualToFilterOperatorComponent_1 = tslib_1.__decorate([
        Component({
            providers: [
                {
                    provide: FilterOperatorBase,
                    useExisting: forwardRef(function () { return GreaterOrEqualToFilterOperatorComponent_1; }) // tslint:disable-line:no-forward-ref
                }
            ],
            selector: 'kendo-filter-gte-operator',
            template: ""
        }),
        tslib_1.__metadata("design:paramtypes", [LocalizationService])
    ], GreaterOrEqualToFilterOperatorComponent);
    return GreaterOrEqualToFilterOperatorComponent;
    var GreaterOrEqualToFilterOperatorComponent_1;
}(FilterOperatorBase));
export { GreaterOrEqualToFilterOperatorComponent };
