// @progress/kendo-angular-grid/dist/es/filtering/operators/is-empty-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { FilterOperatorBase } from './filter-operator.base';
import { LocalizationService } from '@progress/kendo-angular-l10n';
/**
 * Represents the `IsEmpty` (**Is empty**) filter operator.
 */
var IsEmptyFilterOperatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IsEmptyFilterOperatorComponent, _super);
    function IsEmptyFilterOperatorComponent(localization) {
        return _super.call(this, "isempty", localization) || this;
    }
    IsEmptyFilterOperatorComponent_1 = IsEmptyFilterOperatorComponent;
    IsEmptyFilterOperatorComponent = IsEmptyFilterOperatorComponent_1 = tslib_1.__decorate([
        Component({
            providers: [
                {
                    provide: FilterOperatorBase,
                    useExisting: forwardRef(function () { return IsEmptyFilterOperatorComponent_1; }) // tslint:disable-line:no-forward-ref
                }
            ],
            selector: 'kendo-filter-isempty-operator',
            template: ""
        }),
        tslib_1.__metadata("design:paramtypes", [LocalizationService])
    ], IsEmptyFilterOperatorComponent);
    return IsEmptyFilterOperatorComponent;
    var IsEmptyFilterOperatorComponent_1;
}(FilterOperatorBase));
export { IsEmptyFilterOperatorComponent };
