// @progress/kendo-angular-grid/dist/es/filtering/operators/is-not-null-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { FilterOperatorBase } from './filter-operator.base';
import { LocalizationService } from '@progress/kendo-angular-l10n';
/**
 * Represents the `IsNotNull` (**Is not null**) filter operator.
 */
var IsNotNullFilterOperatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IsNotNullFilterOperatorComponent, _super);
    function IsNotNullFilterOperatorComponent(localization) {
        return _super.call(this, "isnotnull", localization) || this;
    }
    IsNotNullFilterOperatorComponent_1 = IsNotNullFilterOperatorComponent;
    IsNotNullFilterOperatorComponent = IsNotNullFilterOperatorComponent_1 = tslib_1.__decorate([
        Component({
            providers: [
                {
                    provide: FilterOperatorBase,
                    useExisting: forwardRef(function () { return IsNotNullFilterOperatorComponent_1; }) // tslint:disable-line:no-forward-ref
                }
            ],
            selector: 'kendo-filter-isnotnull-operator',
            template: ""
        }),
        tslib_1.__metadata("design:paramtypes", [LocalizationService])
    ], IsNotNullFilterOperatorComponent);
    return IsNotNullFilterOperatorComponent;
    var IsNotNullFilterOperatorComponent_1;
}(FilterOperatorBase));
export { IsNotNullFilterOperatorComponent };
