// @progress/kendo-angular-grid/dist/es/filtering/operators/is-not-empty-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { FilterOperatorBase } from './filter-operator.base';
import { LocalizationService } from '@progress/kendo-angular-l10n';
/**
 * Represents the `IsNotEmpty` (**Is not empty**) filter operator.
 */
var IsNotEmptyFilterOperatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IsNotEmptyFilterOperatorComponent, _super);
    function IsNotEmptyFilterOperatorComponent(localization) {
        return _super.call(this, "isnotempty", localization) || this;
    }
    IsNotEmptyFilterOperatorComponent_1 = IsNotEmptyFilterOperatorComponent;
    IsNotEmptyFilterOperatorComponent = IsNotEmptyFilterOperatorComponent_1 = tslib_1.__decorate([
        Component({
            providers: [
                {
                    provide: FilterOperatorBase,
                    useExisting: forwardRef(function () { return IsNotEmptyFilterOperatorComponent_1; }) // tslint:disable-line:no-forward-ref
                }
            ],
            selector: 'kendo-filter-isnotempty-operator',
            template: ""
        }),
        tslib_1.__metadata("design:paramtypes", [LocalizationService])
    ], IsNotEmptyFilterOperatorComponent);
    return IsNotEmptyFilterOperatorComponent;
    var IsNotEmptyFilterOperatorComponent_1;
}(FilterOperatorBase));
export { IsNotEmptyFilterOperatorComponent };
