// @progress/kendo-angular-grid/dist/es/filtering/operators/lt-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { FilterOperatorBase } from './filter-operator.base';
import { LocalizationService } from '@progress/kendo-angular-l10n';
/*
 * Represents the `Less` (**Is less than**) numeric filter operator.
 */
var LessFilterOperatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(LessFilterOperatorComponent, _super);
    function LessFilterOperatorComponent(localization) {
        return _super.call(this, "lt", localization) || this;
    }
    LessFilterOperatorComponent_1 = LessFilterOperatorComponent;
    LessFilterOperatorComponent = LessFilterOperatorComponent_1 = tslib_1.__decorate([
        Component({
            providers: [
                {
                    provide: FilterOperatorBase,
                    useExisting: forwardRef(function () { return LessFilterOperatorComponent_1; }) // tslint:disable-line:no-forward-ref
                }
            ],
            selector: 'kendo-filter-lt-operator',
            template: ""
        }),
        tslib_1.__metadata("design:paramtypes", [LocalizationService])
    ], LessFilterOperatorComponent);
    return LessFilterOperatorComponent;
    var LessFilterOperatorComponent_1;
}(FilterOperatorBase));
export { LessFilterOperatorComponent };
