// @progress/kendo-angular-grid/dist/es/filtering/operators/lte-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { FilterOperatorBase } from './filter-operator.base';
import { LocalizationService } from '@progress/kendo-angular-l10n';
/*
 * Represents the `LessOrEqualTo` (**Is less than or equal to**) numeric filter operator.
 */
var LessOrEqualToFilterOperatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(LessOrEqualToFilterOperatorComponent, _super);
    function LessOrEqualToFilterOperatorComponent(localization) {
        return _super.call(this, "lte", localization) || this;
    }
    LessOrEqualToFilterOperatorComponent_1 = LessOrEqualToFilterOperatorComponent;
    LessOrEqualToFilterOperatorComponent = LessOrEqualToFilterOperatorComponent_1 = tslib_1.__decorate([
        Component({
            providers: [
                {
                    provide: FilterOperatorBase,
                    useExisting: forwardRef(function () { return LessOrEqualToFilterOperatorComponent_1; }) // tslint:disable-line:no-forward-ref
                }
            ],
            selector: 'kendo-filter-lte-operator',
            template: ""
        }),
        tslib_1.__metadata("design:paramtypes", [LocalizationService])
    ], LessOrEqualToFilterOperatorComponent);
    return LessOrEqualToFilterOperatorComponent;
    var LessOrEqualToFilterOperatorComponent_1;
}(FilterOperatorBase));
export { LessOrEqualToFilterOperatorComponent };
