// @progress/kendo-angular-grid/dist/es/filtering/operators/not-contains-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { FilterOperatorBase } from './filter-operator.base';
import { LocalizationService } from '@progress/kendo-angular-l10n';
/**
 * Represents the `DoesNotContain` (**Does not contain**) filter operator.
 */
var DoesNotContainFilterOperatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(DoesNotContainFilterOperatorComponent, _super);
    function DoesNotContainFilterOperatorComponent(localization) {
        return _super.call(this, "doesnotcontain", localization) || this;
    }
    DoesNotContainFilterOperatorComponent_1 = DoesNotContainFilterOperatorComponent;
    DoesNotContainFilterOperatorComponent = DoesNotContainFilterOperatorComponent_1 = tslib_1.__decorate([
        Component({
            providers: [
                {
                    provide: FilterOperatorBase,
                    useExisting: forwardRef(function () { return DoesNotContainFilterOperatorComponent_1; }) // tslint:disable-line:no-forward-ref
                }
            ],
            selector: 'kendo-filter-not-contains-operator',
            template: ""
        }),
        tslib_1.__metadata("design:paramtypes", [LocalizationService])
    ], DoesNotContainFilterOperatorComponent);
    return DoesNotContainFilterOperatorComponent;
    var DoesNotContainFilterOperatorComponent_1;
}(FilterOperatorBase));
export { DoesNotContainFilterOperatorComponent };
