// @progress/kendo-angular-grid/dist/es/localization/custom-messages.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { LocalizationService } from '@progress/kendo-angular-l10n';
import { Messages } from './messages';
/**
 * Custom component messages override default component messages.
 */
var CustomMessagesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(CustomMessagesComponent, _super);
    function CustomMessagesComponent(service) {
        var _this = _super.call(this) || this;
        _this.service = service;
        return _this;
    }
    CustomMessagesComponent_1 = CustomMessagesComponent;
    Object.defineProperty(CustomMessagesComponent.prototype, "override", {
        get: function () {
            return true;
        },
        enumerable: true,
        configurable: true
    });
    CustomMessagesComponent = CustomMessagesComponent_1 = tslib_1.__decorate([
        Component({
            providers: [
                {
                    provide: Messages,
                    useExisting: forwardRef(function () { return CustomMessagesComponent_1; }) // tslint:disable-line:no-forward-ref
                }
            ],
            selector: 'kendo-grid-messages',
            template: ""
        }),
        tslib_1.__metadata("design:paramtypes", [LocalizationService])
    ], CustomMessagesComponent);
    return CustomMessagesComponent;
    var CustomMessagesComponent_1;
}(Messages));
export { CustomMessagesComponent };
