// @progress/kendo-angular-grid/dist/es2015/rendering/common/col-group.component.js
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
import { DetailTemplateDirective } from '../details/detail-template.directive';
import { columnsToRender } from '../../columns/column-common';
/**
 * @hidden
 */
let ColGroupComponent = class ColGroupComponent {
    /**
     * @hidden
     */
    constructor() {
        this.columns = [];
        this.groups = [];
    }
    get columnsToRender() {
        return columnsToRender(this.columns);
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Array)
], ColGroupComponent.prototype, "columns", void 0);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Array)
], ColGroupComponent.prototype, "groups", void 0);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", DetailTemplateDirective)
], ColGroupComponent.prototype, "detailTemplate", void 0);
ColGroupComponent = tslib_1.__decorate([
    Component({
        selector: '[kendoGridColGroup]',
        template: `
    <ng-template [ngIf]="true">
        <col [class.k-group-col]="true" *ngFor="let g of groups" />
        <col [class.k-hierarchy-col]="true" *ngIf="detailTemplate?.templateRef"/>
        <col *ngFor="let column of columnsToRender" [style.width.px]="column.width"/>
    </ng-template>
    `
    })
], ColGroupComponent);
export { ColGroupComponent };
