// @progress/kendo-angular-grid/dist/es2015/editing/remove-command.directive.js
import * as tslib_1 from "tslib";
import { Directive, HostListener, HostBinding, Inject, Optional, ElementRef, Renderer2 as Renderer } from '@angular/core';
import { Button } from '@progress/kendo-angular-buttons';
import { RTL } from '@progress/kendo-angular-l10n';
import { EditService } from './edit.service';
import { CELL_CONTEXT } from '../rendering/common/cell-context';
/**
 * Represents the `remove` command of the Grid.
 *
 * You can apply this directive to any `button` element inside a
 * [`CommandColumnComponent`]({% slug api_grid_commandcolumncomponent %}).
 *
 * When an associated button with the directive is clicked, the
 * [`remove` event]({% slug api_grid_gridcomponent %}#toc-remove)
 * is triggered. For more information, refer to the [editing example]({% slug editing_grid %}).
 *
 * @example
 * ```ts-no-run
 * <kendo-grid>
 *   <kendo-grid-command-column title="command">
 *     <ng-template kendoGridCellTemplate>
 *       <button kendoGridRemoveCommand>Remove row</button>
 *     </ng-template>
 *   </kendo-grid-command-column>
 * </kendo-grid>
 * ```
 * > When the row *is* in the edit mode, the button with the `kendoGridRemoveCommand` is automatically hidden.
 */
let RemoveCommandDirective = class RemoveCommandDirective extends Button {
    constructor(editService, cellContext, element, renderer, rtl) {
        super(element, renderer, null, rtl);
        this.editService = editService;
        this.rowIndex = cellContext.rowIndex;
    }
    /**
     * @hidden
     */
    click() {
        this.editService.remove(this.rowIndex);
    }
    /**
     * @hidden
     */
    get visible() {
        return this.editService.isEdited(this.rowIndex) ? 'none' : '';
    }
    /**
     * @hidden
     */
    get buttonClass() {
        return true;
    }
    /**
     * @hidden
     */
    get commandClass() {
        return true;
    }
};
tslib_1.__decorate([
    HostListener('click'),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", []),
    tslib_1.__metadata("design:returntype", void 0)
], RemoveCommandDirective.prototype, "click", null);
tslib_1.__decorate([
    HostBinding('style.display'),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [])
], RemoveCommandDirective.prototype, "visible", null);
tslib_1.__decorate([
    HostBinding('class.k-button'),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [])
], RemoveCommandDirective.prototype, "buttonClass", null);
tslib_1.__decorate([
    HostBinding('class.k-grid-remove-command'),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [])
], RemoveCommandDirective.prototype, "commandClass", null);
RemoveCommandDirective = tslib_1.__decorate([
    Directive({
        selector: '[kendoGridRemoveCommand]'
    }),
    tslib_1.__param(1, Inject(CELL_CONTEXT)),
    tslib_1.__param(4, Optional()), tslib_1.__param(4, Inject(RTL)),
    tslib_1.__metadata("design:paramtypes", [EditService, Object, ElementRef,
        Renderer, Boolean])
], RemoveCommandDirective);
export { RemoveCommandDirective };
