// @progress/kendo-angular-grid/dist/es2015/rendering/toolbar/toolbar.component.js
import * as tslib_1 from "tslib";
import { Component, HostBinding } from '@angular/core';
import { GridComponent } from "../../grid.component";
/**
 * @hidden
 */
let ToolbarComponent = class ToolbarComponent {
    constructor(grid) {
        this.grid = grid;
    }
    get classNames() {
        return 'k-header k-grid-toolbar';
    }
    get toolbarTemplateRef() {
        return this.grid.toolbarTemplate ? this.grid.toolbarTemplate.templateRef : undefined;
    }
};
tslib_1.__decorate([
    HostBinding('class'),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [])
], ToolbarComponent.prototype, "classNames", null);
ToolbarComponent = tslib_1.__decorate([
    Component({
        selector: 'kendo-grid-toolbar',
        template: `
        <ng-template
            *ngIf="toolbarTemplateRef"
            [ngTemplateOutlet]="toolbarTemplateRef">
        </ng-template>
    `
    }),
    tslib_1.__metadata("design:paramtypes", [GridComponent])
], ToolbarComponent);
export { ToolbarComponent };
