// @progress/kendo-angular-excel-export/dist/es/columns/column-group.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, SkipSelf, Host, Optional } from '@angular/core';
import { ColumnBase } from './column-base';
/**
 * Represents the column-group component of the Kendo UI ExcelExport component.
 */
var ColumnGroupComponent = /** @class */ (function (_super) {
    tslib_1.__extends(ColumnGroupComponent, _super);
    function ColumnGroupComponent(parent) {
        var _this = _super.call(this, parent) || this;
        _this.parent = parent;
        return _this;
    }
    ColumnGroupComponent_1 = ColumnGroupComponent;
    ColumnGroupComponent = ColumnGroupComponent_1 = tslib_1.__decorate([
        Component({
            providers: [
                {
                    provide: ColumnBase,
                    useExisting: forwardRef(function () { return ColumnGroupComponent_1; }) // tslint:disable-line:no-forward-ref
                }
            ],
            selector: 'kendo-excelexport-column-group',
            template: ""
        }),
        tslib_1.__param(0, SkipSelf()), tslib_1.__param(0, Host()), tslib_1.__param(0, Optional()),
        tslib_1.__metadata("design:paramtypes", [ColumnBase])
    ], ColumnGroupComponent);
    return ColumnGroupComponent;
    var ColumnGroupComponent_1;
}(ColumnBase));
export { ColumnGroupComponent };
