// @progress/kendo-angular-grid/dist/es2015/grouping/group-indicator.component.js
import * as tslib_1 from "tslib";
import { Component, HostBinding, Input, Output, EventEmitter, ChangeDetectionStrategy } from '@angular/core';
import { GroupInfoService } from './group-info.service';
/**
 * @hidden
 */
let GroupIndicatorComponent = class GroupIndicatorComponent {
    constructor(groupInfoService) {
        this.groupInfoService = groupInfoService;
        this.directionChange = new EventEmitter();
        this.remove = new EventEmitter();
    }
    get groupIndicatorClass() {
        return true;
    }
    get title() {
        return this.groupInfoService.groupTitle(this.group);
    }
    get dir() {
        return this.group.dir ? this.group.dir : "asc";
    }
    toggleDirection() {
        this.directionChange.emit({
            dir: this.dir === "asc" ? "desc" : "asc",
            field: this.group.field
        });
        return false;
    }
    removeDescriptor() {
        this.remove.emit({
            dir: this.group.dir,
            field: this.group.field
        });
        return false;
    }
};
tslib_1.__decorate([
    Output(),
    tslib_1.__metadata("design:type", EventEmitter)
], GroupIndicatorComponent.prototype, "directionChange", void 0);
tslib_1.__decorate([
    Output(),
    tslib_1.__metadata("design:type", EventEmitter)
], GroupIndicatorComponent.prototype, "remove", void 0);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Object)
], GroupIndicatorComponent.prototype, "group", void 0);
tslib_1.__decorate([
    HostBinding("class.k-group-indicator"),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [])
], GroupIndicatorComponent.prototype, "groupIndicatorClass", null);
GroupIndicatorComponent = tslib_1.__decorate([
    Component({
        changeDetection: ChangeDetectionStrategy.OnPush,
        selector: '[kendoGroupIndicator]',
        template: `
        <a href="#" class="k-link" tabindex="-1" (click)="toggleDirection()">
            <span class="k-icon"
                [class.k-i-sort-asc-sm]="dir === 'asc'"
                [class.k-i-sort-desc-sm]="dir === 'desc'"></span>
            {{title}}</a>
        <a class="k-button k-button-icon k-bare" tabindex="-1" (click)="removeDescriptor()">
            <span class="k-icon k-i-group-delete"></span>
        </a>
    `
    }),
    tslib_1.__metadata("design:paramtypes", [GroupInfoService])
], GroupIndicatorComponent);
export { GroupIndicatorComponent };
