// @progress/kendo-angular-grid/dist/es2015/filtering/operators/contains-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { LocalizationService } from '@progress/kendo-angular-l10n';
import { FilterOperatorBase } from './filter-operator.base';
/**
 * Represents the `Contains` (**Contains**) filter operator.
 */
let ContainsFilterOperatorComponent = ContainsFilterOperatorComponent_1 = class ContainsFilterOperatorComponent extends FilterOperatorBase {
    constructor(localization) { super("contains", localization); }
};
ContainsFilterOperatorComponent = ContainsFilterOperatorComponent_1 = tslib_1.__decorate([
    Component({
        providers: [
            {
                provide: FilterOperatorBase,
                useExisting: forwardRef(() => ContainsFilterOperatorComponent_1) // tslint:disable-line:no-forward-ref
            }
        ],
        selector: 'kendo-filter-contains-operator',
        template: ``
    }),
    tslib_1.__metadata("design:paramtypes", [LocalizationService])
], ContainsFilterOperatorComponent);
export { ContainsFilterOperatorComponent };
var ContainsFilterOperatorComponent_1;
