// @progress/kendo-angular-grid/dist/es2015/filtering/operators/gte-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { FilterOperatorBase } from './filter-operator.base';
import { LocalizationService } from '@progress/kendo-angular-l10n';
/**
 * Represents the `GreaterOrEqualTo` (**Is greater than or equal to**) numeric filter operator.
 */
let GreaterOrEqualToFilterOperatorComponent = GreaterOrEqualToFilterOperatorComponent_1 = class GreaterOrEqualToFilterOperatorComponent extends FilterOperatorBase {
    constructor(localization) { super("gte", localization); }
};
GreaterOrEqualToFilterOperatorComponent = GreaterOrEqualToFilterOperatorComponent_1 = tslib_1.__decorate([
    Component({
        providers: [
            {
                provide: FilterOperatorBase,
                useExisting: forwardRef(() => GreaterOrEqualToFilterOperatorComponent_1) // tslint:disable-line:no-forward-ref
            }
        ],
        selector: 'kendo-filter-gte-operator',
        template: ``
    }),
    tslib_1.__metadata("design:paramtypes", [LocalizationService])
], GreaterOrEqualToFilterOperatorComponent);
export { GreaterOrEqualToFilterOperatorComponent };
var GreaterOrEqualToFilterOperatorComponent_1;
