// @progress/kendo-angular-grid/dist/es2015/filtering/operators/is-not-null-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { FilterOperatorBase } from './filter-operator.base';
import { LocalizationService } from '@progress/kendo-angular-l10n';
/**
 * Represents the `IsNotNull` (**Is not null**) filter operator.
 */
let IsNotNullFilterOperatorComponent = IsNotNullFilterOperatorComponent_1 = class IsNotNullFilterOperatorComponent extends FilterOperatorBase {
    constructor(localization) { super("isnotnull", localization); }
};
IsNotNullFilterOperatorComponent = IsNotNullFilterOperatorComponent_1 = tslib_1.__decorate([
    Component({
        providers: [
            {
                provide: FilterOperatorBase,
                useExisting: forwardRef(() => IsNotNullFilterOperatorComponent_1) // tslint:disable-line:no-forward-ref
            }
        ],
        selector: 'kendo-filter-isnotnull-operator',
        template: ``
    }),
    tslib_1.__metadata("design:paramtypes", [LocalizationService])
], IsNotNullFilterOperatorComponent);
export { IsNotNullFilterOperatorComponent };
var IsNotNullFilterOperatorComponent_1;
