// @progress/kendo-angular-grid/dist/es2015/filtering/operators/isnull-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { FilterOperatorBase } from './filter-operator.base';
import { LocalizationService } from '@progress/kendo-angular-l10n';
/**
 * Represents the `IsNull` (**Is null**) filter operator.
 */
let IsNullFilterOperatorComponent = IsNullFilterOperatorComponent_1 = class IsNullFilterOperatorComponent extends FilterOperatorBase {
    constructor(localization) { super("isnull", localization); }
};
IsNullFilterOperatorComponent = IsNullFilterOperatorComponent_1 = tslib_1.__decorate([
    Component({
        providers: [
            {
                provide: FilterOperatorBase,
                useExisting: forwardRef(() => IsNullFilterOperatorComponent_1) // tslint:disable-line:no-forward-ref
            }
        ],
        selector: 'kendo-filter-isnull-operator',
        template: ``
    }),
    tslib_1.__metadata("design:paramtypes", [LocalizationService])
], IsNullFilterOperatorComponent);
export { IsNullFilterOperatorComponent };
var IsNullFilterOperatorComponent_1;
