// @progress/kendo-angular-grid/dist/es2015/filtering/operators/not-contains-filter-operator.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { FilterOperatorBase } from './filter-operator.base';
import { LocalizationService } from '@progress/kendo-angular-l10n';
/**
 * Represents the `DoesNotContain` (**Does not contain**) filter operator.
 */
let DoesNotContainFilterOperatorComponent = DoesNotContainFilterOperatorComponent_1 = class DoesNotContainFilterOperatorComponent extends FilterOperatorBase {
    constructor(localization) { super("doesnotcontain", localization); }
};
DoesNotContainFilterOperatorComponent = DoesNotContainFilterOperatorComponent_1 = tslib_1.__decorate([
    Component({
        providers: [
            {
                provide: FilterOperatorBase,
                useExisting: forwardRef(() => DoesNotContainFilterOperatorComponent_1) // tslint:disable-line:no-forward-ref
            }
        ],
        selector: 'kendo-filter-not-contains-operator',
        template: ``
    }),
    tslib_1.__metadata("design:paramtypes", [LocalizationService])
], DoesNotContainFilterOperatorComponent);
export { DoesNotContainFilterOperatorComponent };
var DoesNotContainFilterOperatorComponent_1;
