// @progress/kendo-angular-grid/dist/es2015/pdf/pdf-command.directive.js
import * as tslib_1 from "tslib";
import { Directive, HostListener, HostBinding } from '@angular/core';
import { PDFService } from './pdf.service';
/**
 * Represents the **Export to PDF** command of the Grid.
 *
 * You can apply this directive to any `button` element inside a
 * [`ToolbarTemplate`]({% slug api_grid_commandcolumncomponent %}).
 *
 * When the user clicks a button associated with the directive, the
 * [`pdfExport`]({% slug api_grid_gridcomponent %}#toc-pdfexport) event
 * fires. For more information, refer to the [PDF export example]({% slug pdfexport_grid %}).
 *
 * @example
 * ```ts-no-run
 * <kendo-grid>
 *      <ng-template kendoGridToolbarTemplate>
 *          <button kendoGridPDFCommand>Export to PDF</button>
 *      </ng-template>
 *      <kendo-grid-pdf fileName="Grid.pdf">
 *      </kendo-grid-pdf>
 * </kendo-grid>
 * ```
 */
let PDFCommandDirective = class PDFCommandDirective {
    constructor(pdfService) {
        this.pdfService = pdfService;
    }
    /**
     * @hidden
     */
    click() {
        this.pdfService.exportClick.emit();
    }
    /**
     * @hidden
     */
    get buttonClass() {
        return true;
    }
    /**
     * @hidden
     */
    get pdfClass() {
        return true;
    }
};
tslib_1.__decorate([
    HostListener('click'),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", []),
    tslib_1.__metadata("design:returntype", void 0)
], PDFCommandDirective.prototype, "click", null);
tslib_1.__decorate([
    HostBinding('class.k-button'),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [])
], PDFCommandDirective.prototype, "buttonClass", null);
tslib_1.__decorate([
    HostBinding('class.k-grid-pdf'),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [])
], PDFCommandDirective.prototype, "pdfClass", null);
PDFCommandDirective = tslib_1.__decorate([
    Directive({
        selector: '[kendoGridPDFCommand]'
    }),
    tslib_1.__metadata("design:paramtypes", [PDFService])
], PDFCommandDirective);
export { PDFCommandDirective };
