// @progress/kendo-angular-inputs/dist/es2015/slider/localization/custom-messages.component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef } from '@angular/core';
import { LocalizationService } from '@progress/kendo-angular-l10n';
import { Messages } from './messages';
/**
 * Custom component messages override default component messages.
 */
let SliderCustomMessagesComponent = SliderCustomMessagesComponent_1 = class SliderCustomMessagesComponent extends Messages {
    constructor(service) {
        super();
        this.service = service;
    }
    get override() {
        return true;
    }
};
SliderCustomMessagesComponent = SliderCustomMessagesComponent_1 = tslib_1.__decorate([
    Component({
        providers: [
            {
                provide: Messages,
                useExisting: forwardRef(() => SliderCustomMessagesComponent_1) // tslint:disable-line:no-forward-ref
            }
        ],
        selector: 'kendo-slider-messages',
        template: ``
    }),
    tslib_1.__metadata("design:paramtypes", [LocalizationService])
], SliderCustomMessagesComponent);
export { SliderCustomMessagesComponent };
var SliderCustomMessagesComponent_1;
