// @progress/kendo-angular-grid/dist/es2015/editing-directives/in-cell-editing.directive.js
import * as tslib_1 from "tslib";
import { Directive, Input } from '@angular/core';
import { EditingDirectiveBase } from './editing-directive-base';
import { GridComponent } from '../grid.component';
import { LocalDataChangesService } from '../editing/local-data-changes.service';
import { markAllAsTouched } from './utils';
/**
 * A directive which encapsulates the editing operations of the Grid when using the in-cell editing with Reactive Forms.
 */
let InCellEditingDirective = class InCellEditingDirective extends EditingDirectiveBase {
    constructor(grid, localDataChangesService) {
        super(grid, localDataChangesService);
        this.grid = grid;
        this.localDataChangesService = localDataChangesService;
    }
    // Need mixin
    createModel(args) {
        return this.createFormGroup(args);
    }
    saveModel({ dataItem, formGroup, isNew }) {
        if (!formGroup.dirty && !isNew) {
            return;
        }
        if (formGroup.valid) {
            this.editService.assignValues(dataItem, formGroup.value);
            return dataItem;
        }
        markAllAsTouched(formGroup);
    }
    /**
     * @hidden
     */
    ngOnInit() {
        super.ngOnInit();
        this.subscriptions
            .add(this.grid.cellClick.subscribe(this.cellClickHandler.bind(this)))
            .add(this.grid.cellClose.subscribe(this.cellCloseHandler.bind(this)));
    }
    removeHandler(args) {
        super.removeHandler(args);
        this.grid.cancelCell();
    }
    cellClickHandler(args) {
        if (!args.isEdited && args.type !== 'contextmenu') {
            this.grid.editCell(args.rowIndex, args.columnIndex, this.createFormGroup(args));
        }
    }
    cellCloseHandler(args) {
        const { formGroup, dataItem } = args;
        if (!formGroup.valid) {
            args.preventDefault();
        }
        else if (formGroup.dirty) {
            this.editService.assignValues(dataItem, formGroup.value);
            this.editService.update(dataItem);
        }
    }
};
tslib_1.__decorate([
    Input('kendoGridInCellEditing'),
    tslib_1.__metadata("design:type", Function)
], InCellEditingDirective.prototype, "createFormGroup", void 0);
InCellEditingDirective = tslib_1.__decorate([
    Directive({
        selector: '[kendoGridInCellEditing]'
    }),
    tslib_1.__metadata("design:paramtypes", [GridComponent, LocalDataChangesService])
], InCellEditingDirective);
export { InCellEditingDirective };
