// @progress/kendo-angular-inputs/dist/es2015/textbox/textbox.directive.js
import * as tslib_1 from "tslib";
import { Directive, ElementRef, EventEmitter, HostBinding, Output, Renderer2 } from '@angular/core';
/**
 * Represents the Kendo UI TextBox directive for the Inputs components for Angular.
 * Used to style the textbox of any `input` element.
 *
 * @example
 * ```ts-no-run
 * <input kendoTextBox />
 * <input kendoTextBox type="email" />
 * <input kendoTextBox type="password" />
 * ```
 */
let TextBoxDirective = class TextBoxDirective {
    constructor(renderer, inputElement) {
        this.renderer = renderer;
        this.inputElement = inputElement;
        /**
         * Fires each time the user focuses the `input` element.
         */
        this.onFocus = new EventEmitter(); //tslint:disable-line:no-output-rename
        /**
         * Fires each time the `input` element gets blurred.
         */
        this.onBlur = new EventEmitter(); //tslint:disable-line:no-output-rename
        this.listeners = [];
    }
    get hostClass() {
        return true;
    }
    /**
     * Gets the value of the `input` element.
     */
    get value() {
        return this.inputElement.nativeElement.value;
    }
    get id() {
        return this.inputElement.nativeElement.id;
    }
    set id(id) {
        this.renderer.setAttribute(this.inputElement.nativeElement, 'id', id);
    }
    ngAfterViewInit() {
        const input = this.inputElement.nativeElement;
        this.listeners = [
            this.renderer.listen(input, 'focus', () => this.onFocus.emit()),
            this.renderer.listen(input, 'blur', () => this.onBlur.emit())
        ];
    }
    ngOnDestroy() {
        this.listeners.forEach(listener => listener());
    }
};
tslib_1.__decorate([
    HostBinding('class.k-textbox'),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [])
], TextBoxDirective.prototype, "hostClass", null);
tslib_1.__decorate([
    Output('focus'),
    tslib_1.__metadata("design:type", EventEmitter)
], TextBoxDirective.prototype, "onFocus", void 0);
tslib_1.__decorate([
    Output('blur'),
    tslib_1.__metadata("design:type", EventEmitter)
], TextBoxDirective.prototype, "onBlur", void 0);
TextBoxDirective = tslib_1.__decorate([
    Directive({
        selector: 'input[kendoTextBox]'
    }),
    tslib_1.__metadata("design:paramtypes", [Renderer2,
        ElementRef])
], TextBoxDirective);
export { TextBoxDirective };
