// @progress/kendo-angular-inputs/dist/es2015/textbox/textbox-container.component.js
import * as tslib_1 from "tslib";
import { ChangeDetectorRef, ContentChild, Component, ElementRef, EventEmitter, HostBinding, Input, Inject, Optional, Renderer2, isDevMode } from '@angular/core';
import { RTL } from '@progress/kendo-angular-l10n';
import { TextBoxDirective } from './textbox.directive';
import { TextAreaDirective } from './textarea.directive';
import { NgControl } from '@angular/forms';
import { guid } from '../common/dom-utils';
const isFunction = (x) => Object.prototype.toString.call(x) === '[object Function]';
/**
 * Represents the Kendo UI TextBoxContainer component for Angular.
 * Provides floating labels to `input` elements.
 *
 * @example
 * ```ng-template
 * <kendo-textbox-container floatingLabel="First name">
 *   <input kendoTextBox />
 * </kendo-textbox-container>
 * ```
 */
let TextBoxContainerComponent = class TextBoxContainerComponent {
    constructor(elementRef, renderer, changeDetectorRef, rtl) {
        this.elementRef = elementRef;
        this.renderer = renderer;
        this.changeDetectorRef = changeDetectorRef;
        /**
         * @hidden
         */
        this.focused = false;
        /**
         * @hidden
         */
        this.empty = true;
        /**
         * @hidden
         */
        this.invalid = false;
        this._subscriptions = [];
        this.direction = rtl ? 'rtl' : 'ltr';
        this.renderer.removeAttribute(this.elementRef.nativeElement, "id");
    }
    get hostClasses() {
        return true;
    }
    get textareaElementClass() {
        return !!this.textarea;
    }
    get focusedClass() {
        return this.focused;
    }
    get emptyClass() {
        return this.empty;
    }
    get invalidClass() {
        return this.invalid;
    }
    /**
     * @hidden
     */
    ngAfterContentInit() {
        if (!this.formControl && !this.textbox && !this.textarea) {
            if (isDevMode()) {
                throw new Error("The TextBoxContainer requires an element with the kendoTextBox or kendoTextArea directive" +
                    " or a forms-bound component to function properly.");
            }
            return;
        }
        const control = this.textbox || this.textarea || this.formControl.valueAccessor;
        const setFocus = (isFocused) => () => {
            this.focused = isFocused;
            this.updateState();
            this.changeDetectorRef.markForCheck();
        };
        if (control.onFocus instanceof EventEmitter) {
            const focus = control.onFocus.subscribe(setFocus(true));
            this._subscriptions.push(focus);
        }
        if (control.onBlur instanceof EventEmitter) {
            const blur = control.onBlur.subscribe(setFocus(false));
            this._subscriptions.push(blur);
        }
        const updateState = () => this.updateState();
        updateState();
        if (this.formControl) {
            const s = this.formControl.valueChanges.subscribe(updateState);
            this._subscriptions.push(s);
        }
        if ('id' in control) {
            if (this.id && control.id) {
                // input wins
                this.id = control.id;
            }
            else if (this.id) {
                control.id = this.id;
            }
            else if (control.id) {
                this.id = control.id;
            }
            else {
                const id = "_" + guid();
                control.id = id;
                this.id = id;
            }
        }
    }
    /**
     * @hidden
     */
    ngOnDestroy() {
        this._subscriptions.forEach(s => s.unsubscribe());
        this._subscriptions = [];
    }
    updateState() {
        const formControl = this.formControl;
        const empty = value => {
            // zero is not an empty value (e.g. NumericTextBox)
            if (value === 0) {
                return false;
            }
            // empty arrays are an empty value (e.g. MultiSelect)
            if (Array.isArray(value) && !value.length) {
                return true;
            }
            return !value;
        };
        if (formControl) {
            const valueAccessor = formControl.valueAccessor;
            if (isFunction(valueAccessor.isEmpty)) {
                this.empty = valueAccessor.isEmpty();
            }
            else {
                this.empty = empty(formControl.value);
            }
            this.invalid = formControl.invalid && (formControl.touched || formControl.dirty);
        }
        else if (this.textarea) {
            this.empty = empty(this.textarea.value);
        }
        else {
            this.empty = empty(this.textbox.value);
        }
        this.changeDetectorRef.markForCheck();
    }
};
tslib_1.__decorate([
    HostBinding('class.k-textbox-container'),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [])
], TextBoxContainerComponent.prototype, "hostClasses", null);
tslib_1.__decorate([
    HostBinding('class.k-textarea-wrapper'),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [])
], TextBoxContainerComponent.prototype, "textareaElementClass", null);
tslib_1.__decorate([
    HostBinding('class.k-state-focused'),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [])
], TextBoxContainerComponent.prototype, "focusedClass", null);
tslib_1.__decorate([
    HostBinding('class.k-state-empty'),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [])
], TextBoxContainerComponent.prototype, "emptyClass", null);
tslib_1.__decorate([
    HostBinding('class.k-state-invalid'),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [])
], TextBoxContainerComponent.prototype, "invalidClass", null);
tslib_1.__decorate([
    HostBinding('attr.dir'),
    tslib_1.__metadata("design:type", String)
], TextBoxContainerComponent.prototype, "direction", void 0);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String)
], TextBoxContainerComponent.prototype, "id", void 0);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String)
], TextBoxContainerComponent.prototype, "floatingLabel", void 0);
tslib_1.__decorate([
    ContentChild(TextBoxDirective),
    tslib_1.__metadata("design:type", TextBoxDirective)
], TextBoxContainerComponent.prototype, "textbox", void 0);
tslib_1.__decorate([
    ContentChild(TextAreaDirective),
    tslib_1.__metadata("design:type", TextAreaDirective)
], TextBoxContainerComponent.prototype, "textarea", void 0);
tslib_1.__decorate([
    ContentChild(NgControl),
    tslib_1.__metadata("design:type", NgControl)
], TextBoxContainerComponent.prototype, "formControl", void 0);
TextBoxContainerComponent = tslib_1.__decorate([
    Component({
        selector: 'kendo-textbox-container',
        template: `
        <ng-content></ng-content>
        <label *ngIf="floatingLabel" [for]="id" class="k-label">{{ floatingLabel }}</label>
    `
    }),
    tslib_1.__param(3, Optional()), tslib_1.__param(3, Inject(RTL)),
    tslib_1.__metadata("design:paramtypes", [ElementRef,
        Renderer2,
        ChangeDetectorRef, Boolean])
], TextBoxContainerComponent);
export { TextBoxContainerComponent };
